src/components/ng2-map.component.ts

```diff
@@ -114,7 +114,7 @@ export class Ng2MapComponent implements OnChanges, OnDestroy, AfterViewInit {
       window['ng2MapRef'].forEach( ng2MapRef => {
         ng2MapRef.zone.run(function() { ng2MapRef.componentFn(); });
       });
-      window['ng2MapRef'] = [];
+      window['ng2MapRef'].splice(0, window['ng2MapRef'].length);
```

Comment:
By window['ng2MapRef'] = [], we will miss the map object setted by function initializeMap(). Then, we cannot get the map object with the window handler.